tests/test_1.py

```diff
@@ -52,37 +52,32 @@
 }
 
 
-def castles_number(heights):
-    start_height, current_height = None, None
-    castles = 0
-
+def get_castles_number_1(heights):
     if len(heights) == 0:
-        raise Exception
+        raise Exception('Border is empty')
     if len(heights) == 1:
         return 1
 
+    castles = 0
+    current_height = heights[0]
+
     for i in range(len(heights)):
-        if current_height is None:
-            current_height = heights[i]
 
         if heights[i] != current_height:
-            if (start_height is None) or (heights[i] < current_height and start_height < current_height) or \
-                    (heights[i] > current_height and start_height > current_height):
+            if ('start_height' not in locals()
+                    or heights[i] < current_height and start_height < current_height
+                    or heights[i] > current_height and start_height > current_height):
```

Comment:
Why did you removed parentheses around conditions connected with `and`?
Parentheses make whole condition more explicit and readable.
```
or (heights[i] < current_height and start_height < current_height)
```

Your code works without parentheses due different operator precedence: https://docs.python.org/3/reference/expressions.html#operator-precedence
If you don't know about operator precedences (I'm sure you don't) you do not have the right to skip parentheses.